### PR TITLE
[docs][ENG-8521] Rearrange information about VM specs

### PR DESCRIPTION
**Pull request by khamilowicz**
# Why

Docs should use data from `eas-build` repo (https://github.com/expo/eas-build/pull/250) to generate hardware and vm specs.

# Deploy Plan

1. https://github.com/expo/eas-build/pull/250
2. https://github.com/expo/expo/pull/22601
3. https://github.com/expo/universe/pull/12701

### CODE REVIEW

File: docs/pages/build-reference/infrastructure.mdx [hunk 1]
```diff
@@ -148,17 +148,14 @@ Android builders run on virtual machines in an isolated environment. Every build
 
 ## iOS build server configurations
 
-iOS builder VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build gets its own fresh macOS VM:
+iOS builder VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build gets its own fresh macOS VM.
```
Comment (szdziedzic) [inline docs/pages/build-reference/infrastructure.mdx:151]:
```suggestion
iOS builder VMs run on Mac Mini hosts in an isolated environment. Every build gets its own fresh macOS VM.
```

I think that 8.1 is no longer true (?). Let's drop this number.

File: docs/pages/build-reference/infrastructure.mdx [hunk 2]
```diff
@@ -31,9 +31,9 @@ When selecting an image for the build you can use the full name provided below o
 
 Android builders run on virtual machines in an isolated environment. Every build gets its own dedicated VM instance.
 
-- Build resources:
-  - [`medium`](eas-json/#resourceclass-1): 4 CPU, 16 GB RAM ([`n2-standard-4`](https://cloud.google.com/compute/docs/general-purpose-machines#n2_machines) Google Cloud machine type)
-  - [`large`](eas-json/#resourceclass-1): 8 CPU, 32 GB RAM ([`n2-standard-8`](https://cloud.google.com/compute/docs/general-purpose-machines#n2_machines) Google Cloud machine type)
+- Hardware:
+  - 4 CPU, 16 GB RAM (n2-standard-4 Google Cloud machine type)
+  - 8 CPU, 32 GB RAM (n2-standard-8 Google Cloud machine type)
```
Comment (szdziedzic) [inline docs/pages/build-reference/infrastructure.mdx:36]: Link to pricing page descriptions from here https://github.com/expo/universe/pull/12408

File: docs/pages/build-reference/infrastructure.mdx [hunk 3]
```diff
@@ -148,17 +148,14 @@ Android builders run on virtual machines in an isolated environment. Every build
...
 
-iOS builder VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build gets its own fresh macOS VM:
+iOS builder VMs run on Mac Mini 8.1 hosts in an isolated environment. Every build gets its own fresh macOS VM.
+[Learn more about it here](eas-json/#resourceclass-2)
 
 - Hardware:
   - Intel: Intel(R) Core(TM) i7-8700B CPU (6 cores/12 threads), 64 GB RAM
   - M1: 3.2GHz 8-Core M1 (4 performance and 4 efficiency cores), 16 GB RAM
   - M2: 3.4GHz 12-Core M2 (8 performance and 4 efficiency cores), 24 GB RAM
   - M2 Pro: 3.4GHz 10-Core M2 Pro (6 performance and 4 efficiency cores), 32 GB RAM
```
Comment (szdziedzic) [inline docs/pages/build-reference/infrastructure.mdx:158]: Same as above

File: docs/public/static/schemas/unversioned/eas-json-build-android-schema.js [hunk 4]
```diff
@@ -17,7 +17,13 @@ export default [
...
     enum: ['default', 'medium', 'large'],
     description: [
-      'The Android-specific resource class that will be used to run this build. [Learn more](../../build-reference/infrastructure#android-build-server-configurations)',
+      'The Android-specific resource class that will be used to run this build.',
+      '',
+      'Android builders run on virtual machines in an isolated environment. Every build gets its own dedicated VM instance.',
+      '',
+      'Build resources:',
+      '- `medium` 4 CPU, 16 GB RAM (n2-standard-4 Google Cloud machine type)',
+      '- `large` 8 CPU, 32 GB RAM (n2-standard-8 Google Cloud machine type)',
```
Comment (szdziedzic) [inline docs/public/static/schemas/unversioned/eas-json-build-android-schema.js:26]: Same as above

File: docs/public/static/schemas/unversioned/eas-json-build-ios-schema.js [hunk 5]
```diff
@@ -35,11 +35,17 @@ export default [
...
     description: [
-      'The iOS-specific resource class that will be used to run this build. [Learn more](../../build-reference/infrastructure#ios-build-server-configurations)',
+      'The iOS-specific resource class that will be used to run this build.',
       '- For SDK version >= 45 or React Native version >= 0.71.0 `default` maps to `m-medium`, otherwise it maps to `intel-medium`',
       '- For SDK version >= 45 or React Native version >= 0.71.0 `medium` maps to `m-medium`, otherwise it maps to `intel-medium`',
       '',
+      'Build resources:',
+      '- `intel-medium` 3 cores, 12 GB RAM, 2 builder VMs per host',
+      '- `m-medium` 2 cores, 8 GB RAM, 2 builder VMs per host',
+      '- `large` 4 cores, 12 GB RAM, 2 builder VMs per host (for builds running on M2 Pro hosts) or 4 cores, 22 GB RAM, 1 builder VM per host (for builds running on M2 hosts)',
```
Comment (szdziedzic) [inline docs/public/static/schemas/unversioned/eas-json-build-ios-schema.js:45]: same as above

**Review by szdziedzic — CHANGES_REQUESTED** [review]
I think that idea behind this task was to make our infra docs easier to maintain, meaning that ideally, we wouldn't need to change hardware/resource class descriptions in multiple places.

If we are going with adding resource class/hardware descriptions to the Website https://github.com/expo/universe/pull/12408 I would just link to the pricing page from these docs, so we don't need to edit both docs and the pricing page every time we add new hardware.

File: docs/pages/build-reference/infrastructure.mdx [hunk 6]
```diff
@@ -6,6 +6,33 @@ description: Learn about the current build server infrastructure when using EAS.
...
+          return (
+            <li key={name}>
+              {cpu}, {memory} ({description})
+            </li>
+          );
+        })
+      )}
+    </ul>
+  );
+}
```
Comment (Simek) [inline docs/pages/build-reference/infrastructure.mdx:35]: Let's extract this component to separate file inside `~/ui/components`.

File: docs/pages/build-reference/infrastructure.mdx [hunk 7]
```diff
@@ -6,6 +6,33 @@ description: Learn about the current build server infrastructure when using EAS.
...
 import { Collapsible } from '~/ui/components/Collapsible';
+import al from '@mdx-js/react';
+import { values, keys } from 'lodash';
+
+import resourceSpecs from '~/public/static/resource-specs.json';
+
+export const {
+  hardware: hardwareSpecs,
+  resources: { ios, android },
+} = resourceSpecs.data;
```
Comment (Simek) [inline docs/pages/build-reference/infrastructure.mdx:17]: Would be nice to made this object restructure during JSON generation.

File: docs/scripts/resource-specs-sync.mjs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  if (!response.ok) {
+    console.error(`Failed to fetch resource specs from ${resourceSpecsSource}`);
+    console.error(`Response status: ${response.status}`);
+    console.error(`Response body: ${await response.text()}`);
+    return;
+  }
+
+  const data = await response.text();
+
+  // TODO: is public the correct place for this?
```
Comment (szdziedzic) [inline docs/scripts/resource-specs-sync.mjs:20]: I think this is a good question for the docs code owners 😄

**Review by szdziedzic — APPROVED** [review]
Seems good

**Pull request merged**